src/OpenLoco/src/Map/TileManager.cpp

```diff
@@ -1267,6 +1268,86 @@ namespace OpenLoco::World::TileManager
         return totalCost;
     }
 
+    // 0x004C4C28
+    uint32_t adjustWaterHeight(World::Pos2 pos, SmallZ targetHeight, std::set<World::Pos3, LessThanPos3>& removedBuildings, uint8_t flags)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Construction);
+        GameCommands::setPosition(World::Pos3(pos.x + World::kTileSize / 2, pos.y + World::kTileSize / 2, targetHeight * kMicroToSmallZStep));
```

Comment:
targetHeight is in SmallZ so this is the wrong constant (although it does have the same value as the correct constant)